File: backend/app/schemas/calculator.py
from typing import Literal
from pydantic import BaseModel, Field


class CalcRequest(BaseModel):
    site_id: str
    time_to_rain_min: int = Field(..., ge=0, description="Минут до дождя")
    mix_temp_c: int = Field(140, ge=100, le=160, description="Температура смеси на выходе °C")
    paver_width_m: float = Field(7.0, ge=2.0, le=7.5, description="Ширина укладчика м")
    layer_type: Literal["standard", "thin"] = "standard"


class CalcResponse(BaseModel):
    site_id: str
    time_to_rain_min: int
    compaction_time_min: int
    available_paving_min: int
    max_tonnage_t: float
    trucks_needed: int
    recommendation: str
    can_start: bool

File: frontend/src/types/calculator.ts
export interface CalcRequest {
  site_id: string;
  time_to_rain_min: number;
  mix_temp_c: number;
  paver_width_m: number;
  layer_type: 'standard' | 'thin';
}

export interface CalcResponse {
  site_id: string;
  time_to_rain_min: number;
  compaction_time_min: number;
  available_paving_min: number;
  max_tonnage_t: number;
  trucks_needed: number;
  recommendation: string;
  can_start: boolean;
}
